Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles.cs

```diff
@@ -1,220 +1,216 @@
-            return logger;
+            var fileTransporter = new FileTransporter(logger, _batchContext, executionId);
+            var result = fileTransporter.Run(cancellationToken);
+
+            if (options.ThrowErrorOnFail && !result.Success)
+                throw new Exception($"SFTP transfer failed: {result.UserResultMessage}. " +
+                                    $"Latest operations: \n{GetLogLines(transferSink.GetBufferedLogMessages())}");
+
+            if (options.OperationLog)
+                result.OperationsLog = GetLogDictionary(transferSink.GetBufferedLogMessages());
+
+            return new Result(result);
         }
+    }
+
+    /// <summary>
+    /// Fetches log lines in case of exception.
+    /// </summary>
+    /// <param name="buffer"></param>
+    /// <returns></returns>
+    private static string GetLogLines(IEnumerable<Tuple<DateTimeOffset, string>> buffer)
+    {
+        try
+        {
+            return string.Join("\n", buffer.Select(x => x.Item1 == DateTimeOffset.MinValue ? "..." : $"{x.Item1:HH:mm:ssZ}: {x.Item2}"));
 
-        private static IDictionary<string, string> GetLogDictionary(IList<Tuple<DateTimeOffset, string>> entries)
+        }
+        catch (Exception e)
         {
-            const string dateFormat = "yyyy-MM-dd HH:mm:ss.f0Z";
+            return $"Error while creating log: \n{e.ToString()}";
+        }
+    }        
 
-            try
-            {
-                return entries
-                    .Where(e => e?.Item2 != null) // Filter out nulls
-                    .ToLookup(
-                        x => x.Item1.ToString(dateFormat))
-                    .ToDictionary(
-                        x => x.Key,
-                        x => string.Join("\n", x.Select(k => k.Item2)));
-            }
-            catch (Exception e)
+    private static SFTPLogger InitializeSFTPLogger(ILogger notificationLogger)
+    {
+        var logger = new SFTPLogger(notificationLogger);
+        return logger;
+    }
+
+    private static IDictionary<string, string> GetLogDictionary(IList<Tuple<DateTimeOffset, string>> entries)
+    {
+        const string dateFormat = "yyyy-MM-dd HH:mm:ss.f0Z";
+
+        try
+        {
+            return entries
+                .Where(e => e?.Item2 != null) // Filter out nulls
+                .ToLookup(
+                    x => x.Item1.ToString(dateFormat))
+                .ToDictionary(
+                    x => x.Key,
+                    x => string.Join("\n", x.Select(k => k.Item2)));
+        }
+        catch (Exception e)
+        {
+            return new Dictionary<string, string>
             {
-                return new Dictionary<string, string>
-                {
-                    { DateTimeOffset.Now.ToString(dateFormat), $"Error while creating operation log: \n{e.ToString()}" }
-                };
-            }
+                { DateTimeOffset.Now.ToString(dateFormat), $"Error while creating operation log: \n{e.ToString()}" }
```

Comment:
If you are using string interpretation,  if value is not a string, compiler will automatically call the objects ToString method. You can change `{e.ToString()}` to `{e}`.

Reply:
Fixed.